tools/serverpod_cli/test/database/extentions/migration_with_many_to_many_relation_pgsql_code_generation_test.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/models/stateful_analyzer.dart';
+import 'package:serverpod_cli/src/database/create_definition.dart';
+import 'package:serverpod_cli/src/generator/code_generation_collector.dart';
+import 'package:serverpod_cli/src/test_util/builders/model_source_builder.dart';
+import 'package:serverpod_service_client/serverpod_service_client.dart';
+import 'package:test/test.dart';
+
+import '../../generator/dart/client_code_generator/class_constructors_test.dart';
+
+void main() {
```

Comment:
Seeing the code causing the problem, I think this should also be reproducible with a simpler 2 table setup. One refers to the other, and when we rename the "target table" it should occur as well (as the "Course" part in this example seems to be not relevant).